Remove all the "active turfs at round start" warnings from Box

ojacobson opened the pull request:
Some chucklefuck had set a bunch of tiles to have `temperature = 273.15` rather than leaving the `temperature` attribute unset; I've removed the attribute from those tiles. I've also made tiles exposed to space use `../airless` subtypes more consistently, including a couple of new airless types (cult floors, magma).
- [x] Remove the hilarious Dream Maker puke
- [x] Adjacent tiles with different air, z1-6
- [x] Air tiles adjacent to space, z1-6

ojacobson committed:
Remove `temperature` setting from a bunch of turfs.

This silences a large swath of the "active turf on round start" warnings in the log.

Jordie0608 commented:
You might want to try doing this with a text editor, Notepad++ is a good choice.

ojacobson committed:
Make rooms exposed to space more space-y by setting up atmos consistently.

Space-exposed turfs are now at a steady state at roundstart, and fully in vacuum. Bring internals and a hardsuit.

ojacobson commented on code/game/turfs/simulated/floor/plasteel_floor.dm:
```diff
@@ -122,6 +122,11 @@
+	oxygen = 0
+	nitrogen = 0
+	temperature = TCMB
+
```
haha whups

ojacobson commented:
uuuuughhhh the wizards' den

It's enclosed by force walls, which aren't considered by the code that detects active atmos tiles at roundstart, but does hold in air. As a result, a bunch of the active tile warnings are from parts of the wizards' den which it thinks are exposed to space, but actually aren't. Not sure what to do here.

* Surround the wizards' den with fake space tiles and fill them with air, and enclose that in walls?
* Go the other direction and start wizards in a vacuum? (lolno)
* Replace the tiles under the force walls with walls that look like floors, so the wizards' den is actually sealed, and hope nobody notices that they can't move out of the den into space by destroying the force walls?
* Teach the active-tiles-at-round-start code (and atmos code generally) how to tell that air will not spread into a force wall?
* Redesign the wizards' den to be sealed, but still wizardy and cool?
* Ignore it? (It's cluttering up my debugging output right now.)

I'm weakly inclined to put fake-floor walls under the forcewall tiles, since it seems like the least work, but I'm not sold on it.

optimumtact commented:
Fix the code that detects

ojacobson commented:
sadness

MrStonedOne commented:
oj, there is an object proc thats used for detecting this by the atmos code.

so override it on false walls,

ojacobson commented:
@MrStonedOne Thanks, I'll dig through this and fix the detect loop.

ojacobson committed:
Forcewalls and mimewalls now block atmos properly.

They were blocking atmos before, I think? This makes the "active atmos tiles"
code at startup do the right thing, though.

ojacobson commented:
There, I think this is good to go. I'll do the other stations in separate pull requests; doing this one was tiring.

Aranclanos commented on code/controllers/subsystem/air.dm:
```diff
@@ -193,6 +193,7 @@ var/datum/subsystem/air/SSair
 					if(!T.air.check_turf_total(enemy_tile))
 						T.excited = 1
 						active_turfs |= T
+						break
```
whats this

ojacobson replied:
Previously, a single atmos tile floating in space would be detected four times. (This actually happened; there was a single tile of plasteel out in the middle of Z=2 somewhere.) Now it'll be detected once, consistent with the case where tiles are adjacent to other atmos tiles with differing gas mixes.

Aranclanos replied:
yeah good find

Metacide commented:
@ojacobson - fixed most of MetaStation myself, so no need to work on that if you were considering it. Thanks all the same, particularly for finding the temperature var issue. Made fixing it much simpler. :+1:
